python/cuml/utils/input_utils.py

```diff
@@ -223,22 +186,102 @@ def input_to_dev_array(X, order='F', deepcopy=False,
+    * cuDF Series - returns by reference or a deep copy depending on
+        `deepcopy`.
+    * Numpy array - returns a copy in device always
+    * cuda array interface compliant array (like Cupy) - returns a
+        reference unless `deepcopy`=True.
+    * numba device array - returns a reference unless deepcopy=True
 
-def convert_dtype(X, to_dtype=np.float32):
+    Parameters
+    ----------
+
+    X : cuDF.DataFrame, cuDF.Series, numba array, NumPy array or any
+        cuda_array_interface compliant array like CuPy or pytorch.
+
+    order: string (default: 'F')
+        Whether to return a F-major or C-major array. Used to check the order
+        of the input. If fail_on_order=True method will raise ValueError,
+        otherwise it will convert X to be of order `order`.
+
+    deepcopy: boolean (default: False)
+        Set to True to always return a deep copy of X.
+
+    check_dtype: np.dtype (default: False)
+        Set to a np.dtype to throw an error if X is not of dtype `check_dtype`.
+
+    convert_to_dtype: np.dtype (default: False)
+        Set to a dtype if you want X to be converted to that dtype if it is
+        not that dtype already.
+
+    check_cols: int (default: False)
+        Set to an int `i` to check that input X has `i` columns. Set to False
+        (default) to not check at all.
+
+    check_rows: boolean (default: False)
+        Set to an int `i` to check that input X has `i` columns. Set to False
+        (default) to not check at all.
+
+    fail_on_order: boolean (default: False)
+        Set to True if you want the method to raise a ValueError if X is not
+        of order `order`.
+
+    Returns
+    -------
+    `inp_array`: namedtuple('inp_array', 'array pointer n_rows n_cols dtype')
+
+        A new device array if the input was not a numba device
+        array. It is a reference to the input X if it was a numba device array
+        or cuda array interface compliant (like cupy)
+
+    """
+
+    ary_tuple = input_to_cuml_array(X,
+                                    order=order,
+                                    deepcopy=deepcopy,
+                                    check_dtype=check_dtype,
+                                    convert_to_dtype=convert_to_dtype,
+                                    check_cols=check_cols,
+                                    check_rows=check_rows,
+                                    fail_on_order=fail_on_order)
+
+    return inp_array(array=cuda.as_cuda_array(ary_tuple.array),
+                     pointer=ary_tuple.array.ptr,
+                     n_rows=ary_tuple.n_rows,
+                     n_cols=ary_tuple.n_cols,
+                     dtype=ary_tuple.dtype)
+
+
+def convert_dtype(X, to_dtype=np.float32, legacy=True):
     """
     Convert X to be of dtype `dtype`
 
     Supported float dtypes for overflow checking.
     Todo: support other dtypes if needed.
     """
 
+    # temporarily importing here, until github issue #xx reorganizing utils
```

Comment:
Reference issue. Maybe one issue to encapsulate a bunch of these same/similar comments?